src/metabase/api/search.clj

```diff
@@ -515,12 +516,19 @@
    table_db_id  (s/maybe su/IntGreaterThanZero)
    models       (s/maybe models-schema)}
   (api/check-valid-page-params mw.offset-paging/*limit* mw.offset-paging/*offset*)
-  (search (search-context
-           q
-           archived
-           table_db_id
-           models
-           mw.offset-paging/*limit*
-           mw.offset-paging/*offset*)))
+  (let [start-time (System/currentTimeMillis)
+        results    (search (search-context
+                            q
+                            archived
+                            table_db_id
+                            models
+                            mw.offset-paging/*limit*
+                            mw.offset-paging/*offset*))
+        duration   (- (System/currentTimeMillis) start-time)]
+    (when (and (nil? models)
```

Comment:
it's not clear to me why we want this guard, could you explain? and maybe put it into the code as well

Reply:
The spec says to track "global search". Any of these arguments limit search in a way that is not global.

Reply:
Added a comment [Test for only tracking global searches](https://github.com/metabase/metabase/pull/31467/commits/fb06f1ba33bbad2b0dec194f876aaddf224c6330)